### PR TITLE
glooctl uninstall removes ratelimitconfig crd

### PR DESCRIPTION
**Pull request by npolshakova**
# Description

The RateLimitConfig CRD was not getting removed after running `glooctl uninstall --all`. Adds the ratelimitconfig to the gloo crd names to be used when no Gloo release object exists.  

# Context

Fixes https://github.com/solo-io/gloo/issues/4010

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/4010

### CODE REVIEW

File: projects/gloo/cli/pkg/cmd/install/install_suite_test.go [hunk 1]
```diff
@@ -25,7 +30,7 @@ var dir string
 var file, values1, values2 string
 
 const (
-	unitTestingTaggedVersion = "vunit-testing"
+	unitTestingTaggedVersion = "v1.6.0-vunit-testing"
```
Comment (saiskee) [inline projects/gloo/cli/pkg/cmd/install/install_suite_test.go:33]:
```suggestion
	unitTestingTaggedVersion = "v1.6.0-unit-testing"
```

Comment (npolshakova) [reply projects/gloo/cli/pkg/cmd/install/install_suite_test.go:33]: Nice catch!

**Comment by saiskee** [comment]
Had a nit but looks like you just addressed that! Looks good to me otherwise.

File: projects/gloo/cli/pkg/cmd/install/install_suite_test.go [hunk 2]
```diff
@@ -70,6 +75,30 @@ settings:
...
+			runtimeObj, err := runtime.Decode(unstructured.UnstructuredJSONScheme, jsn)
+			Expect(err).NotTo(HaveOccurred())
+
+			// get crd name from yaml
+			resource := runtimeObj.(*unstructured.Unstructured)
+			Expect(resource.GetName()).NotTo(BeNil())
+			crdNames = append(crdNames, resource.GetName())
+		}
+	}
+	Expect(crdNames).To(ContainElements(install.GlooCrdNames))
```
Comment (mlholland) [inline projects/gloo/cli/pkg/cmd/install/install_suite_test.go:101]: The `crdNames` array should filter out for all CRD's that don't end in 'solo.io', then perform a member equality check against `install.GlooCrdNames`. That way we can ensure that there are neither any forgotten CRDs, nor any extra CRD's that haven't been added to the CRD list yet. This'll protect againt the most likely future failure case: newly added CRD's not getting added to `install.GlooCrdNames`.

Comment (npolshakova) [reply projects/gloo/cli/pkg/cmd/install/install_suite_test.go:101]: Yeah, I think you're right we should check that extra crds haven't been added. Why do we need to filter out crds that don't end in 'solo.io'? Do you think there's a case when a non-solo.io crd would get added to /gloo/install/helm/crds that we wouldn't want to uninstall?

**Review by mlholland — APPROVED** [review]

**Pull request merged**